apis/core/v1beta1/common/common.go

```diff
@@ -55,3 +89,28 @@ func FeatureFlagConfigurationId(namespace, name string) string {
 func FeatureFlagConfigMapKey(namespace, name string) string {
 	return fmt.Sprintf("%s.flagd.json", FeatureFlagConfigurationId(namespace, name))
 }
+
+func RemoveDuplicateEnvVars(input []corev1.EnvVar) []corev1.EnvVar {
+	list := map[string]corev1.EnvVar{}
+	out := []corev1.EnvVar{}
+	for _, item := range input {
+		list[item.Name] = item
+	}
+	for _, val := range list {
+		out = append(out, val)
+	}
+	return out
+}
+
+func RemoveDuplicatesGeneric[T comparable](input []T) []T {
```

Comment:
should we just maintain the generic one? 🤔

Reply:
this would be ideal, but if you consider `EnvVar` struct, we are not removing duplications (my bad, will adapt the naming). Let's take an example:

```yaml
- name: key1
   value: val1
- name: key1
   value: val2
```

These two items are not duplications, as the values are different, but in our use-case, we cannot have 2x the same key in `envVars`, therefore we need to store the key with the value only once. In this case taking the latest value, as this is the approach that is used when merging structs in OFO

Reply:
Then we should call it `RemoveDuplicatesSlice`, the generic aspect is forced by the compiler anyway via syntax :p